### PR TITLE
Add documentation and unittests regarding multisets

### PR DESCRIPTION
**Pull request by wilzbach**
Resubmission of https://github.com/dlang/phobos/pull/5627 due to GH's annoying hook issue (they send the wrong sha after a rebase).

### CODE REVIEW

File: std/algorithm/setops.d [hunk 1]
```diff
@@ -3,6 +3,20 @@
...
 It contains generic algorithms that implement set operations.
 
+The functions $(LREF multiwayMerge), $(LREF multiwayUnion), $(LREF setDifference),
+$(LREF setIntersection), $(LREF setSymmetricDifference) expect a range of sorted
+ranges as input.
+
+All the algorithms, except (LREF multiwayMerge) and (LREF mutiwayUnion),
+are generalized to accept as input not only sets but also multisets:
+$(UL
+    $(UI when all ranges passed as input are sets, the result is going to be
```
Comment (wilzbach) [inline std/algorithm/setops.d]:
```
/dev/shm/dtest/work/repo/dlang.org/web/phobos-prerelease/std_algorithm_setops.html:157:<ul>    <!--UNDEFINED MACRO: "UI"-->
/dev/shm/dtest/work/repo/dlang.org/web/phobos-prerelease/std_algorithm_setops.html:158:    <!--UNDEFINED MACRO: "UI"-->
```

@RazvanN7 as we currently can't rebase PRs without breaking GH hooks, could you please fix this on your branch (should be `LI`) and resubmit a PR? Thanks!

File: std/algorithm/setops.d [hunk 2]
```diff
@@ -870,14 +941,17 @@ as a range of ranges and each is assumed to be sorted by $(D
 less). Computation is done lazily, one union element at a time.
 `multiwayUnion(ror)` is functionally equivalent to `multiwayMerge(ror).uniq`.
 
+`multiwayUnion` returns a set regardless of the type of the input ranges
+(sets or multisets).
```
Comment (andralex) [inline std/algorithm/setops.d]: We don't distinguish sets/multisets by "type" so the sentence is misleading. Replace with:

"The output of `multiwayUnion` has no duplicates even when its inputs contain duplicates."

File: std/algorithm/setops.d [hunk 3]
```diff
@@ -1019,6 +1117,12 @@ Lazily computes the intersection of two or more input ranges $(D
 ranges). The ranges are assumed to be sorted by $(D less). The element
 types of the ranges must have a common type.
 
+If all input ranges are sets (i.e. without duplicated elements),
+the resulting range is going to be a set, otherwise the result
+is going to be a multiset. In the case of multisets, the range with
```
Comment (andralex) [inline std/algorithm/setops.d]: No, as shown above intersecting multisets may produce a set. Just delete the first sentence in this paragraph; the second contains all needed information.

File: std/algorithm/setops.d [hunk 4]
```diff
@@ -616,6 +636,22 @@ void largestPartialIntersection
...
+    // multiset
+    double[][] x =
+    [
+        [1, 1, 1, 1, 4, 7, 8],
+        [1, 7],
+        [1, 7, 8],
+        [4, 7],
+        [7]
+    ];
+    auto y = new Tuple!(double, uint)[2];
```
Comment (RazvanN7) [inline std/algorithm/setops.d]: why not auto? the rest of the examples are like this.

**Comment by RazvanN7** [comment]
Close this

File: std/algorithm/setops.d (hunk 4 above)
Comment (andralex) [inline std/algorithm/setops.d]: This works: `int[10] a;`
This doesn't: `auto a = int[10];`

Comment (andralex) [inline std/algorithm/setops.d]: note that crucially the statically sized array avoids memory allocation

**Pull request closed without merging**